src/requests.rs

```diff
@@ -0,0 +1,339 @@
+use crate::login::PlantBuddyRole;
+use crate::management::User;
+use base64::{
+    alphabet,
+    engine::{self, general_purpose},
+    Engine as _,
+};
+use log::info;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::hash::Hash;
+use std::ops::Range;
+
+const ENDPOINT: &str = "https://pb.mfloto.com/v1/";
+
+/// Represents a temporary user returned by the login API.
+#[derive(Deserialize, Debug)]
+struct TempUser {
+    id: u32,
+    name: String,
+    role: u64,
+}
+
+/// Represents a temporary user used to create a new user.
+#[derive(Deserialize, Debug, Serialize, Clone)]
+pub struct TempCreationUser {
```

Comment:
And then another one, this time with a password and the role is a u64 when it needs to represent three states?

Why do we not use PlantBuddyRole dirrectly and convert to u64 when needed? 

Also couldn't the three Struct Problem be better solved through composition, making it more elegant. We could use one larger user Struct and give references to it via Arc to our component's keeping our credentials in sync and efficient and every component can access every field they need. This would also what the login method gives back